opamp-client/src/main/java/io/opentelemetry/opamp/client/internal/request/service/HttpRequestService.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opamp.client.internal.request.service;
+
+import io.opentelemetry.opamp.client.internal.connectivity.http.HttpErrorException;
+import io.opentelemetry.opamp.client.internal.connectivity.http.HttpSender;
+import io.opentelemetry.opamp.client.internal.connectivity.http.RetryAfterParser;
+import io.opentelemetry.opamp.client.internal.request.Request;
+import io.opentelemetry.opamp.client.internal.request.delay.AcceptsDelaySuggestion;
+import io.opentelemetry.opamp.client.internal.request.delay.PeriodicDelay;
+import io.opentelemetry.opamp.client.internal.response.OpampServerResponseError;
+import io.opentelemetry.opamp.client.internal.response.Response;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import opamp.proto.AgentToServer;
+import opamp.proto.ServerErrorResponse;
+import opamp.proto.ServerErrorResponseType;
+import opamp.proto.ServerToAgent;
+
+public final class HttpRequestService implements RequestService {
+  private final HttpSender requestSender;
+  private final ScheduledExecutorService executorService;
```

Comment:
perhaps there should be a comment that this must be a single threaded executor and that requests shouldn't be sent concurrently